src/server/routes/contract/subscriptions/backfillContractSubscription.ts

```diff
@@ -0,0 +1,124 @@
+import { Static, Type } from "@sinclair/typebox";
+import { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import { getAddress } from "thirdweb";
+import { getContractSubscription } from "../../../../db/contractSubscriptions/getContractSubscription";
+import { enqueueProcessEventLogs } from "../../../../worker/queues/processEventLogsQueue";
+import { createCustomError } from "../../../middleware/error";
+import { standardResponseSchema } from "../../../schemas/sharedApiSchemas";
+
+const MAX_BACKFILL_BLOCKS = 1000;
+
+const responseSchema = Type.Object({
+  result: Type.Object({
+    message: Type.Literal(`Blocks were added for backfilling!`),
+    status: Type.Literal("success"),
+  }),
+});
+
+responseSchema.example = {
+  result: {
+    message: `Blocks were added for backfilling!`,
+    status: "success",
+  },
+};
+
+const paramsSchema = Type.Object({
+  subscriptionId: Type.String({
+    description: "Subscription ID",
+  }),
+});
+
+const bodySchema = Type.Object({
+  fromBlock: Type.Integer({
+    description: "Start block number",
+  }),
+  toBlock: Type.Integer({
+    description: "End block number",
+  }),
+});
+
+export async function backfillContractSubscriptions(fastify: FastifyInstance) {
+  fastify.route<{
+    Reply: Static<typeof responseSchema>;
+    Params: Static<typeof paramsSchema>;
+    Body: Static<typeof bodySchema>;
+  }>({
+    method: "POST",
+    url: "/contract-subscriptions/:subscriptionId/backfill",
+    schema: {
+      summary: "Backfill a contract subscription",
+      description:
+        "Backfill contract subscription data to trigger any missed webhooks or backfill new events added after subscription is updated",
+      tags: ["Contract-Subscriptions"],
+      operationId: "backfillContractSubscription",
+      response: {
+        ...standardResponseSchema,
+        [StatusCodes.OK]: responseSchema,
+      },
+      params: Type.Object({
+        subscriptionId: Type.String({
+          description: "Subscription ID",
+        }),
+      }),
+    },
+    handler: async (request, reply) => {
+      const { subscriptionId } = request.params;
+
+      const contractSubscription = await getContractSubscription({
```

Comment:
Looks like here you are retriving the first data on DB. If we have multiple subs for the same chain, we should get all the subs and add all the contracts from them.